back/lib/tasks/single_use/20240311_migrate_deprecated_title_and_survey_results_widgets.rake

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+namespace :single_use do
+  desc 'Fix existing layouts'
+  task migrate_deprecated_title_and_survey_results_widgets: [:environment] do |_t, _args|
+    content_buildable_type = 'ReportBuilder::Report'
+
+    def resolved_name(node)
+      type = node['type']
+      type['resolvedName'] if type.is_a?(Hash)
+    end
+
+    def survey_question_node(phase, project, question, parent)
+      {
+        'type' => { 'resolvedName' => 'SurveyQuestionResultWidget' },
+        'nodes' => [],
+        'props' => {
+          'phaseId' => phase.id,
+          'projectId' => project.id,
+          'questionId' => question.id
+        },
+        'custom' => {},
+        'hidden' => false,
+        'parent' => parent,
+        'isCanvas' => false,
+        'displayName' => 'SurveyQuestionResultWidget',
+        'linkedNodes' => {}
+      }
+    end
+
+    Tenant.prioritize(Tenant.creation_finalized).each do |tenant|
+      tenant.switch do
+        Rails.logger.info tenant.host
+
+        layouts = ContentBuilder::Layout.where(content_buildable_type: content_buildable_type)
+        layouts.each do |layout|
+          layout.craftjs_json.dup.each do |(node_id, node)|
+            node['custom'] = {} # we previously stored some data here, but it's not used
+
+            case resolved_name(node)
+            when 'TitleMultiloc'
+              node['type']['resolvedName'] = 'TextMultiloc'
+              node['custom'] = {}
+              node['props']['text'].transform_values! { |text| "<h3>#{text}</h3>" }
+            when 'SurveyResultsWidget'
+              phase = Phase.find_by(id: node['props']['phaseId'])
+              project = Project.find_by(id: node['props']['projectId'])
+              if phase.nil? || project.nil?
+                Rails.logger.info "Phase or project not found for SurveyResultsWidget. layout.id: #{layout.id}, projectId: #{node['props']['projectId']} phaseId: #{node['props']['phaseId']}"
+                next
+              end
+
+              # Includes SurveyResultsGeneratorService#visit_* methods
+              # and ignores these fields front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/_deprecated/SurveyResultsWidget/QuestionFilter.tsx:34
```

Comment:
@luucvanderzee This is why I believe `linear_scale` has been supported for 1.5 years since this commit https://github.com/CitizenLabDotCo/citizenlab/commit/d1f753a3b2b6d8a048d4eda5948dc521a16daf99#diff-f519d3e628f000e1fe7fb51de96cb14194d9cefaa2e4bc308d781f53858c1419R31

Reply:
I also just switched to `e9204fdf6ad (2023-08-02 12:12 +0200)`, and I can see linear scale results here 
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/377c77ea-c05e-468e-a2e2-5e83b76a1a92)